tests/utils/component_data_wrapper.py

```diff
@@ -95,15 +137,38 @@ def copy_with_new_timestamp(self, new_timestamp: datetime) -> InverterDataWrappe
 class EvChargerDataWrapper(EVChargerData):
     """Wrapper for the EvChargerData with default arguments."""
 
-    active_power: float = math.nan
-    current_per_phase: Tuple[float, float, float] = field(
-        default_factory=lambda: (math.nan, math.nan, math.nan)
-    )
-    voltage_per_phase: Tuple[float, float, float] = field(
-        default_factory=lambda: (math.nan, math.nan, math.nan)
-    )
-    cable_state: EVChargerCableState = EVChargerCableState.UNSPECIFIED
-    component_state: EVChargerComponentState = EVChargerComponentState.UNSPECIFIED
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        component_id: int,
+        timestamp: datetime,
+        active_power: float = math.nan,
+        current_per_phase: Tuple[float, float, float] | None = None,
+        voltage_per_phase: Tuple[float, float, float] | None = None,
+        cable_state: EVChargerCableState = EVChargerCableState.UNSPECIFIED,
+        component_state: EVChargerComponentState = EVChargerComponentState.UNSPECIFIED,
+    ) -> None:
+        """Initialize the EvChargerDataWrapper.
+
+        This is a wrapper for the EvChargerData with default arguments. The parameters
+        are documented in the EvChargerData class.
+        """
+        super().__init__(
+            component_id=component_id,
+            timestamp=timestamp,
+            active_power=active_power,
+            current_per_phase=(
+                (math.nan, math.nan, math.nan)
+                if not current_per_phase
+                else current_per_phase
+            ),
+            voltage_per_phase=(
+                (math.nan, math.nan, math.nan)
+                if not voltage_per_phase
+                else voltage_per_phase
```

Comment:
Global suggestion for this commit : I think the positive path should be the pattern rather than the other way around

Reply:
in all these cases, it was either the "default or forward value", where default was the different part, so I prioritized that.  But it is readable for me with your style as well, so changed.